### PR TITLE
Remove the loading icon div and replace it by a pure css solution using :after.

### PR DESCRIPTION
**Pull request by calixteman**
This way we don't have a lot of useless divs and we let the css engine handle the creation/destruction of the :after pseudo-element. It'll help to slightly improve performance when zooming.

### CODE REVIEW

**Comment by calixteman** [comment]
Here is a profile from Firefox beta without the patch:
https://share.firefox.dev/3JAzElB
and one with the patch:
https://share.firefox.dev/3Hs7vdF

File: web/pdf_page_view.js [hunk 1]
```diff
@@ -1001,6 +992,7 @@ class PDFPageView {
       if (isCanvasHidden) {
         canvas.hidden = false;
         isCanvasHidden = false;
+        canvasWrapper.classList.add("loading");
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:995]: Why is this necessary now, since I don't think it should be?
(If this is actually needed, that feels like a bug elsewhere...)

File: web/pdf_page_view.js [hunk 2]
```diff
@@ -1085,7 +1077,8 @@ class PDFPageView {
     return result;
   }
 
-  paintOnSvg(wrapper) {
+  paintOnSvg() {
+    const wrapper = this.#canvasWrapper;
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:1081]: This needs to be moved *below* the `PDFJSDev` checks, since otherwise it's not getting excluded in Firefox.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]

File: web/pdf_page_view.js (hunk 1 above)
Comment (calixteman) [reply web/pdf_page_view.js]: I added an explanation here:
https://github.com/mozilla/pdf.js/pull/15992/files#diff-c48e3561004f5db8f11d5ebab2fd661591222ba911cb4173fbced15f026bac6bR1001

I think it's better to have it in the setTimeout in order to have an almost respected delay.

File: web/pdf_page_view.js [hunk 3]
```diff
@@ -1007,6 +998,12 @@ class PDFPageView {
     canvasWrapper.append(canvas);
     this.canvas = canvas;
 
+    // Adding the loading class is slightly postponed in order to not have
+    // it with loadingIcon.
+    // If we don't do that the visibility of the background is changed but the
+    // transition isn't triggered.
+    setTimeout(() => canvasWrapper.classList.add("loading"), 0);
+
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:1006]: Shouldn't this be placed after e.g. [this line instead](https://github.com/mozilla/pdf.js/pull/15992/files#diff-c48e3561004f5db8f11d5ebab2fd661591222ba911cb4173fbced15f026bac6bR801), to make the loadingIcon work with both canvas and svg mode?

File: web/pdf_page_view.js [hunk 4]
```diff
@@ -236,17 +238,16 @@ class PDFPageView {
 
     switch (state) {
       case RenderingStates.INITIAL:
+        this.#canvasWrapper?.classList.add("loadingIcon");
+        break;
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:242]: Why is this needed, shouldn't we still hide the loadingIcon until rendering is actually started?

Comment (calixteman) [reply web/pdf_page_view.js:242]: The `loadingIcon` is hidden by default:
https://github.com/mozilla/pdf.js/pull/15992/files#diff-e676e52ece9bec7287184fc8ab806d393ca42da7bb3105ac93b4b6cd0d23a3efR161

then adding/removing `loading` will trigger a visibility change.

Comment (Snuffleupagus) [reply web/pdf_page_view.js:242]: When rendering is cancelled we'll set RenderingStates.INITIAL and it seems that we definitely want to *ensure* that the loadingIcon is hidden in that case as well, hence I don't understand why these new lines are introduced here!?

Comment (calixteman) [reply web/pdf_page_view.js:242]: When the rendering is finished, the icon is removed but when the page is zoomed, reset is called and the state is set to INITIAL, loadingIcon is added but invisible and when RUNNING is set then `loading` is added and the icon is made visible.

Comment (Snuffleupagus) [reply web/pdf_page_view.js:242]: With this patch the `loadingIcon` now isn't added until the `draw` method is called, which can happen much later (or not at all) after the `reset` method was invoked.

Hence I don't understand why we simply don't want to *still* do the same thing in the INITIAL/PAUSED cases here!

**Comment by Snuffleupagus** [comment]
Testing the preview it now seems that the `loadingIcon` is displayed *immediately*, and no longer waits the 400 ms as before.

**Review by flodolo — APPROVED** [review]
(feel free to not wait for review if the patch is only removing text)

File: web/pdf_page_view.js [hunk 5]
```diff
@@ -511,16 +507,6 @@ class PDFPageView {
...
-
-    if (!this.loadingIconDiv) {
-      this.loadingIconDiv = document.createElement("div");
-      this.loadingIconDiv.className = "loadingIcon notVisible";
-      this.loadingIconDiv.setAttribute("role", "img");
-      this.l10n.get("loading").then(msg => {
-        this.loadingIconDiv?.setAttribute("aria-label", msg);
-      });
-      div.append(this.loadingIconDiv);
-    }
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js:523]: Quick local testing seem to indicate that https://github.com/mozilla/pdf.js/pull/15992#issuecomment-1413583272 can be fixed by placing the `div.classList.add("loadingIcon");` line here instead of below.

Comment (Snuffleupagus) [reply web/pdf_page_view.js:523]: Why is that a problem, since it'd *considerably* simplify the implementation?

Comment (calixteman) [reply web/pdf_page_view.js:523]: Because the pseudo elements are traversed on restyle.
You can verify that in using the Firefox profiler and in looking at the tab `Marker Chart` at the line `Styles` (I tested with `pdf.pdf`).
Not having the pseudo elements divides by 2 the time spent to restyle when zooming. So from a performance point of view it's better like this.

Anyway if you've some ideas to keep the perf improvement and at the same time to simplify the implementation, I take.

**Review by Snuffleupagus — APPROVED** [review]
r=me, since I'm finding it much nicer with all of this code contained in one method; thank you!
